proto-build/src/transformers.rs

```diff
@@ -312,6 +362,96 @@ pub fn serde_alias_id_with_uppercased(s: ItemStruct) -> ItemStruct {
     syn::ItemStruct { fields, ..s }
 }
 
+fn is_field_gogoproto_nullable(
+    message_name: &str,
+    field_name: &str,
+    descriptor: &protobuf::descriptor::FileDescriptorSet,
+) -> bool {
+    for file in descriptor.file.iter() {
+        for message in file.clone().message_type {
+            // not a message we are searching for
+            if message.name() != message_name {
+                continue;
+            }
+
+            for field in message.clone().field {
+                // not a field we are searching for
+                if field.name() != field_name {
+                    continue;
+                }
+
+                // rust-protobuf can't parse option extensions properly (or i did not find a way for that),
+                // so all options extensions are located in special_fields->unknown_fields after a parsing
+                let sf = field
+                    .options
+                    .special_fields
+                    .unknown_fields()
+                    .get(GOGOPROTO_NULLABLE_EXTENSION_FIELD_NUMBER);
+
+                let is_gogoproto_nullabe = if let Some(v) = sf {
+                    match v {
+                        // Varint(1) => gogoproto.nullable = true
+                        // Varint(0) => gogoproto.nullable = false
+                        // any other enum values (Fixed32, Fixed64, ...) => this means under GOGOPROTO_NULLABLE_EXTENSION_FIELD_NUMBER
+                        // field is contained some other stuff, that is not gogoproto at all (which is highly unlikely)
+                        // let's mark it as unimplemented for now.
+                        // if we really encounter this, we'll come up with some solution
+                        protobuf::UnknownValueRef::Varint(u) => u == 1,
+                        _ => unimplemented!(),
+                    }
+                } else {
+                    false
+                };
+
+                return is_gogoproto_nullabe;
+            }
+        }
+    }
+
+    false
+}
+
+pub fn respect_gogoproto_nullable(
```

Comment:
I wonder if we can have tests for it. would be nice to get to know if some of this fragile logic breaks because of either ours or counterparty's changes to proto tags or `GOGOPROTO_NULLABLE_EXTENSION_FIELD_NUMBER` or something else. for example, apply serialization/deserialization on several proto messages taken from different sources and make sure the results re optionality are as expected

Reply:
> apply serialization/deserialization on several proto messages taken from different sources and make sure the results re optionality are as expected

I don't get, could you elaborate? What exactly do you want to test?